object-filter/src/main/java/org/infinispan/objectfilter/impl/util/ReflectionHelper.java

```diff
@@ -250,13 +252,26 @@ public static PropertyAccessor getAccessor(Class<?> clazz, String propertyName)
       if (propertyName == null || propertyName.length() == 0) {
          throw new IllegalArgumentException("Property name cannot be null or empty");
       }
-      if (propertyName.contains(".")) {
+      if (propertyName.indexOf('.') != -1) {
          throw new IllegalArgumentException("The argument cannot be a nested property name");
       }
+      String propertyNameSuffix = Character.toUpperCase(propertyName.charAt(0)) + propertyName.substring(1);
+
+      Class<?> c = clazz;
+      while (c != null) {
+         PropertyAccessor m = getAccessor(c, propertyName, propertyNameSuffix);
```

Comment:
hahah no no, it's ok.. I know there is no better way to do it. But such constructs are always scary :D